backend/authentication/views/genealogy.py

```diff
@@ -0,0 +1,40 @@
+from rest_framework import views
+from rest_framework.response import Response
+
+from authentication.models import User
+from .profile import HIDDEN_PARENT
+
+class GenealogyView(views.APIView):
+    @staticmethod
+    def chemin_links(chemin):
+        """
+            For a given chemin of n students, returns list of size n-1 of links between each students
+        """
+        chemin_links = []
+        for i in range(len(chemin) - 1):
+            chemin_links.append(chemin[i].relation_with(chemin[i + 1]) + ' de ')
+        return chemin_links
+
+    def get(self, request):
+        eleves = User.objects.all()
+        result = []
+        recherche = False
+        result_string = ""
+        return Response({'eleves': eleves, 'result': result, 'result_string': result_string, 'recherche': recherche})
+
+    def post(self, request):
```

Comment:
If it gives an information and does not modify anything, if should be a `GET`method. In `api/users.ts`, having `getGenealogy` calling `api.post` is a bit weird. Instead you can simply declare a new path with a direct method, and you'll spare the dedicated `APIView`